### PR TITLE
NR-298654: Adding ubuntu 24 support for logs recipe

### PR DESCRIPTION
**Pull request by nr-rkallempudi**

### CODE REVIEW

**Comment by nr-rkallempudi** [comment]
Extending SuperAgent and Infra agent support for logs on following distributions:
1. Ubuntu 24.04
2. Suse 15.*

**Review by danybmx — CHANGES_REQUESTED** [review]
Hi @nr-rkallempudi thanks a lot for the contribution! Although I left some comments this is so nice!

**Review by danybmx — DISMISSED** [review]
👌🏻

File: recipes/newrelic/infrastructure/logs/linux-logs.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ installTargets:
   - type: host
     os: linux
     platformFamily: suse
-    platformVersion: "(15\\.[1-4]|15-SP[1-4])"
+    platformVersion: "(15)\\.*"
```
[the remark this answers is not in the document]
Comment (nr-rkallempudi) [reply recipes/newrelic/infrastructure/logs/linux-logs.yml]: yes. It is added considering that common yaml configuration available at this time for both OTel and Fluent Bit. Plan is there to separate configs for Otel and Fluent Bit so managing would be lot easier.

File: test/definitions/logging/suse155-logs.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            {
+                "id": "nr_infra_suse155",
+                "resource_ids": [
+                    "suse5-logs"
+                ],
+                "provider": "newrelic",
+                "source_repository": "https://github.com/newrelic/open-install-library",
+                "deploy_script_path": "test/deploy/linux/newrelic-cli/install/roles",
+                "params": {
+                    "validate_output": "Logs Integration\\s+\\(installed\\)",
```
Comment (Nandu-pns) [inline test/definitions/logging/suse155-logs.json:28]: It looks you're not specifying the logs integration target to install, could you please check ? 

For reference, please have a look at this - https://github.com/newrelic/open-install-library/blob/main/test/definitions/logging/amazonlinux2-logs.json#L26

File: test/definitions/super-agent/suse/suse155-super-agent.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (Nandu-pns) [inline test/definitions/super-agent/suse/suse155-super-agent.json:1]: Recipes changes are done in the super-agent/logs, hence the test definition configs needs to be created for super-agent logs and not for super-agent.

At the moment the changes are done here `test/definitions/super-agent/` instead these should be done at `test/definitions/super-agent/logs`

Comment (nr-rkallempudi) [reply test/definitions/super-agent/suse/suse155-super-agent.json]: updated PR

**Comment by nr-rkallempudi** [comment]
Following are the effective changes done as part of the PR:
1. Extended Logs support for Ubuntu 24.04
2. Extended Logs support for Suse 15.5
3. Removed installTargets of Suse 15.5 for aarch64 as Fluent Bit support is not available
4. Updated test cases. Tests are not validated on local host. Requested Virtuous team to run the tests before approving the PR

File: test/definitions/super-agent/logs/suse155-super-agent.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            {
+                "id": "nr_super_agent",
+                "resource_ids": [
+                    "host1"
+                ],
+                "provider": "newrelic",
+                "source_repository": "https://github.com/newrelic/open-install-library.git",
+                "deploy_script_path": "test/deploy/linux/newrelic-cli/install-recipe/roles",
+                "params": {
+                    "newrelic_cli_option": "-n logs-integration",
```
Comment (Nandu-pns) [inline test/definitions/super-agent/logs/suse155-super-agent.json:28]: The `newrelic_cli_option` param needs to be updated to `"-n super-agent,logs-integration-super-agent",` and "env_var": "NEW_RELIC_CLI_SKIP_CORE=1" param needs to be added.

An example test config with such params here https://github.com/newrelic/open-install-library/blob/main/test/definitions/super-agent/logs/suse154-super-agent-logs.json#L28-L29

File: test/definitions/super-agent/logs/suse155-super-agent.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                    "recipe_content_url": "https://raw.githubusercontent.com/newrelic/open-install-library/main/recipes/newrelic/infrastructure/super-agent/suse.yml",
+                    "validate_output": "New Relic Super Agent\\s+\\(installed\\)",
```
Comment (Nandu-pns) [inline test/definitions/super-agent/logs/suse155-super-agent.json:30]: `validate_output` should be updated to validate logs integration as below.

`"validate_output": "Logs Integration\\s+\\(installed\\)",`

**Review by danybmx — DISMISSED** [review]

**Comment by Nandu-pns** [comment]
Hi @nr-rkallempudi - Changes looks good to us. I was able to test them out and could see `logs-integration` is being installed successfully on the latest host versions.

**Review by Nandu-pns — DISMISSED** [review]

**Comment by nr-rkallempudi** [comment]
Done validations on Suse 15.4 and Suse 15.5 and identified that Fluent bit is not installing on all AMIs of SLES 15.4 and 15.5. 

Created Jira task to track this one on priority.
https://new-relic.atlassian.net/browse/NR-301797

**Screenshots of successful installation of fluent bit on EC2 hosts:**
Screenshot 2024-08-10 at 2 45 23 PM

Screenshot 2024-08-10 at 2 46 02 PM

File: recipes/newrelic/infrastructure/logs/linux-logs.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nr-rkallempudi) [reply recipes/newrelic/infrastructure/logs/linux-logs.yml]: It is clarified in following conversations that above config is used by only fluent bit. so remove all(.*) versions match and added only supported ones upto 15.5

File: test/definitions/super-agent/logs/suse155-super-agent-logs.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        "Department": "product",
+        "Product": "virtuoso"
+    },
+    "resources": [
+        {
+            "id": "host1",
+            "provider": "aws",
+            "type": "ec2",
+            "size": "t3.nano",
+            "ami_name": "suse-sles-15-sp5-chost-byos-v????????-hvm-ssd-x86_64"
```
Comment (Avinash-Sanpala) [inline test/definitions/super-agent/logs/suse155-super-agent-logs.json:14]: @nr-rkallempudi just want to make sure if this AMI is compatible with the ami instance mentioned in the fluent bit packages of the respective version? Also, If you think this has to be modified upon checking in the fluent bit packages repo, please modify the instance type accordingly similar to the changes done to above logs deploy config files

**Review by Avinash-Sanpala — APPROVED** [review]
LGTM!

**Pull request merged**